src/components/program/ProgramStageSelect.js

```diff
@@ -1,83 +1,85 @@
+import { useDataQuery } from '@dhis2/app-runtime'
 import i18n from '@dhis2/d2-i18n'
 import PropTypes from 'prop-types'
-import React, { Component } from 'react'
-import { connect } from 'react-redux'
-import { loadProgramStages } from '../../actions/programs.js'
+import React, { useCallback, useEffect } from 'react'
 import { SelectField } from '../core/index.js'
 
-class ProgramStageSelect extends Component {
-    static propTypes = {
-        loadProgramStages: PropTypes.func.isRequired,
-        onChange: PropTypes.func.isRequired,
-        className: PropTypes.string,
-        errorText: PropTypes.string,
-        program: PropTypes.object,
-        programStage: PropTypes.object,
-        programStages: PropTypes.object,
-    }
-
-    componentDidUpdate() {
-        const {
-            program,
-            programStage,
-            programStages,
-            loadProgramStages,
-            onChange,
-        } = this.props
+// Load all stages for one program
+const PROGRAM_STAGES_QUERY = {
+    stages: {
+        resource: 'programs',
+        id: ({ id }) => id,
+        params: {
+            fields: ['programStages[id,displayName~rename(name)]'],
```

Comment:
does it work if it is not an array, but a string? I'm not sure what is required.

Reply:
Switched to a string